packages/backend/src/modules/activity2/services/TxsCountProvider.ts

```diff
@@ -0,0 +1,72 @@
+import { Logger } from '@l2beat/backend-tools'
+import { ScalingProjectTransactionApi } from '@l2beat/config'
+import { assert, ProjectId, UnixTime } from '@l2beat/shared-pure'
+import { range } from 'lodash'
+import { Peripherals } from '../../../peripherals/Peripherals'
+import { RpcClient } from '../../../peripherals/rpcclient/RpcClient'
+import { promiseAllPlus } from '../../../tools/queue/promiseAllPlus'
+
+interface TxsCountProviderDeps {
+  logger: Logger
+  peripherals: Peripherals
+  projectId: ProjectId
+  projectConfig: ScalingProjectTransactionApi
+}
+
+export class TxsCountProvider {
+  constructor(private readonly $: TxsCountProviderDeps) {
+    this.$.logger = $.logger.for(this)
```

Comment:
Let's include projectId 
```suggestion
    this.$.logger = $.logger.for(this).tag($.projectId.toString())
```